File: Ligueylou.Server/Dtos/UtilisateurDto.cs
using Ligueylou.Server.Models.Enums;

namespace Ligueylou.Server.Dtos
{
    public class UtilisateurDto
    {
        public Guid Id { get; set; }
        public string Prenom { get; set; }
        public string Nom { get; set; }
        public string? Email { get; set; }
        public SexeEnum? Sexe { get; set; }
        public string? Telephone { get; set; }
        public RoleEnum Role { get; set; }
        public bool Actif { get; set; }
        public DateTime DateCreation { get; set; }

    }
}

File: ligueylou.client/src/app/core/models/UtilisateurDto.ts
import { RoleEnum, SexeEnum } from "./user.enums";

export interface UtilisateurDto {
  id: string;
  prenom: string;
  nom: string;
  email?: string;
  telephone?: string;
  role: RoleEnum;
  sexe?: SexeEnum;
  actif: boolean;
  dateCreation: Date;
  derniereConnexion?: Date;
}

export interface UpdateUtilisateurDto {
  prenom?: string;
  nom?: string;
  email?: string;
  telephone?: string;
  sexe?: SexeEnum;
}
